bndtools.core/src/bndtools/views/repository/RepositoriesView.java

```diff
@@ -613,17 +618,42 @@ public void menuAboutToShow(IMenuManager manager) {
 
                                         description = m.group(4);
                                     }
-                                    Action a = new Action(label) {
+                                    final Action a = new Action(label) {
                                         @Override
                                         public void run() {
-                                            try {
-                                                e.getValue().run();
-                                                if (rp != null && rp instanceof Refreshable)
-                                                    Central.refreshPlugin((Refreshable) rp);
-                                            } catch (Exception e) {
-                                                throw new RuntimeException(e);
-                                            }
-                                            viewer.refresh();
+                                            Job backgroundJob = new Job("Repository Action '" + getText() + "'") {
+
+                                                @Override
+                                                protected IStatus run(IProgressMonitor monitor) {
+                                                    try {
+                                                        e.getValue().run();
+                                                        if (rp != null && rp instanceof Refreshable)
+                                                            Central.refreshPlugin((Refreshable) rp);
+                                                    } catch (final Exception e) {
+                                                        Display.getDefault().asyncExec(new Runnable() {
```

Comment:
Should this really be a dialog? Or just put in the error log? I doubt there is much the user can do here except dismiss the dialog.